docs/src/main/asciidoc/oauth2-guide.adoc

```diff
@@ -0,0 +1,357 @@
+        String helloReply = String.format("hello + %s, isSecure: %s, authScheme: %s", name, ctx.isSecure(), ctx.getAuthenticationScheme());
+        return helloReply;
+    }
+}
+----
+<1> This new endpoint will be located at /secured/roles-allowed
+<2> @RolesAllowed indicates that the given endpoint is accessible by a caller if they have either a "Echoer" or "Subscriber" role assigned.
+
+After you make this addition to your `TokenSecuredResource`, try `curl -v http://127.0.0.1:8080/secured/roles-allowed; echo` to attempt to access the new endpoint. Your output should be:
+
+.curl command for /secured/roles-allowed
+[source,shell]
+----
+$ curl -v http://127.0.0.1:8080/secured/roles-allowed; echo
+*   Trying 127.0.0.1...
+* TCP_NODELAY set
+* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
+> GET /secured/roles-allowed HTTP/1.1
+> Host: 127.0.0.1:8080
+> User-Agent: curl/7.54.0
+> Accept: */*
+>
+< HTTP/1.1 401 Unauthorized
+< Connection: keep-alive
+< Content-Type: text/html;charset=UTF-8
+< Content-Length: 14
+< Date: Sun, 03 Mar 2019 16:32:34 GMT
+<
+* Connection #0 to host 127.0.0.1 left intact
+Not authorized
+----
+
+Excellent, we have not provided any Oauth2 token in the request, so we should not be able to access the endpoint, and we were not. Instead we received an HTTP 401 Unauthorized error. We need to obtain and pass in a valid OAuth2 token to access that endpoint. There are two steps to this, 1) configuring our {extension-name} extension with information on how to validate the token, and 2) generating a matching token with the appropriate claims.
+
+== Configuring the {extension-name} Extension Security Information
+
+In the <<Configuration>> section we introduce the `application.properties` file that affect the {extension-name} extension.
+
+=== Setting up application.properties
+
+ For part A of step 1, create a using-oauth2-rbac/src/main/resources/application.properties with the following content:
+
+.application.properties for TokenSecuredResource
+[source, properties]
+----
+quarkus.oauth2.client-id=client_id
+quarkus.oauth2.client-secret=secret
+quarkus.oauth2.introspection-url=http://oauth-server/introspect
+----
+
+You need to specify the introspection URL of your authentication server and the client_id/client_secret that your application will use to authenticate itself to the authentication server.
+
+The extension will then use these informations to validate the token and recover the information associate with it.
+
+
+=== Generating a token
+
+You need to obtains the token from a standard OAuth2 authentication server (https://www.keycloak.org/[Keycloak] for example) using the token endpoint.
+
+You can find bellow a curl example of such call for a client_credential flow:
+
+[source,shell]
+----
+curl -X POST "http://oauth-server/token?grant_type=client_credentials" \
+-H  "Accept: application/json" -H  "Authorization: Basic Y2xpZW50X2lkOmNsaWVudF9zZWNyZXQ="
+----
+
+It should respond something like that
```

Comment:
It should respond with something like that...

Reply:
done